manifests/service.pp

```diff
@@ -4,31 +4,44 @@
 #
 # @api private
 class puppet_metrics_dashboard::service {
-  service { $puppet_metrics_dashboard::influx_db_service_name:
-    ensure  => running,
-    enable  => true,
-    require => Package['influxdb'],
-  }
-
-  if $puppet_metrics_dashboard::enable_chronograf {
-    service { 'chronograf':
-      ensure  => running,
-      enable  => true,
-      require => [
-        Package['chronograf'],
-        Service[$puppet_metrics_dashboard::influx_db_service_name]
-      ],
+  case $facts['os']['family'] {
+    'Suse': {
+      systemd::unit_file { "${puppet_metrics_dashboard::influx_db_service_name}.service":
+        source => "puppet:///modules/${module_name}/influxdb.service",
+      }
+      ~> service { $puppet_metrics_dashboard::influx_db_service_name:
+        ensure => running,
+        enable => true,
+      }
```

Comment:
I am not going to block on this but would appreciate if you'd use something like this instead of the chaining arrows - I personally find them hard to follow and easy to overlook.

```suggestion
      service { $puppet_metrics_dashboard::influx_db_service_name:
        ensure    => running,
        enable    => true,
        require   => Systemd::Unit_file["${puppet_metrics_dashboard::influx_db_service_name}.service"],
        subscribe => Systemd::Unit_file["${puppet_metrics_dashboard::influx_db_service_name}.service"],
      }
```

Reply:
I always thought that subscribe automatically does the require?

Reply:
I honestly don't know if it does or doesn't ...

Reply:
It does (according to code documentation) https://github.com/puppetlabs/puppet/blob/main/lib/puppet/type.rb#L1652

Reply:
Awesome, thanks for educating me! (sincerely)

Reply:
more precise: https://github.com/puppetlabs/puppet/blob/main/lib/puppet/type.rb#L1646 The direction in is the ordering used in to_edges: https://github.com/puppetlabs/puppet/blob/main/lib/puppet/type.rb#L1580